src/wavesurfer.js

```diff
@@ -92,6 +92,20 @@ var WaveSurfer = {
 
         // Click-to-seek
         this.drawer.on('click', function (e, progress) {
+            var nominalWidth = Math.round(
+                my.getDuration() * my.params.minPxPerSec * my.params.pixelRatio
+            );
+            var parentWidth = my.drawer.getWidth();
+
+            if (!my.params.fillParent && nominalWidth < parentWidth) {
+                var bbox = my.drawer.wrapper.getBoundingClientRect();
+                progress = (e.clientX - bbox.left) * my.params.pixelRatio / nominalWidth;
+
+                if (progress > 1) {
+                    progress = 1;
+                }
+            }
```

Comment:
True, but I think you can take `this.drawer.width` (or `this.width` in the drawer code) instead of calculating `nominalWidth`.